### PR TITLE
RavenDB-8240 Fixed grouping by null

### PR DESCRIPTION
**Pull request by arekpalinski**

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/ReduceIndex/GroupByFieldsRetriever.cs [hunk 1]
```diff
@@ -180,6 +183,14 @@ public override SyntaxNode VisitGroupClause(GroupClauseSyntax node)
                         continue;
                     }
 
+                    if (fieldName[0] == '(' && fieldName.Trim(')').EndsWith("null"))
+                    {
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/ReduceIndex/GroupByFieldsRetriever.cs:187]: `group by item.Gnull` ?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/ReduceIndex/GroupByFieldsRetriever.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/ReduceIndex/GroupByFieldsRetriever.cs:187]: Not sure what you mean. 
- `Gnull` is valid field name identifier
- `group by item.null` isn't valid c# code

Comment (ayende) [reply src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/ReduceIndex/GroupByFieldsRetriever.cs:187]: `group by item.@null` is , and will be compiled as `item.null`, no?

Comment (arekpalinski) [reply src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/ReduceIndex/GroupByFieldsRetriever.cs:187]: oh you meant `@null`, but this case isn't grouping by constant. then it's properly of document / map result

**Pull request merged**